server/session.go

```
@@ -0,0 +1,312 @@
+	session, ok := sessionValue.(SessionWithTools)
+	if !ok {
+		return ErrSessionDoesNotSupportTools
+	}
+
+	sessionTools := session.GetSessionTools()
+	
+	// Create a new map to avoid concurrent modification issues
+	newSessionTools := make(map[string]ServerTool, len(sessionTools)+len(tools))
+	
+	// Copy existing tools
+	if sessionTools != nil {
```

golangci-lint (1.64.8)

S1031: unnecessary nil check around range

(gosimple)